lute/static/js/dict-tabs.js

```diff
@@ -0,0 +1,445 @@
+      const faviconEl = createImg(faviconURL, "dict-btn-fav-img");
+      option.textContent = domain.split("www.").splice(-1)[0];
+      option.prepend(faviconEl);
+      option.dataset.dictId = origIndex;
+      option.dataset.dictExternal = isURLExternal(dict);
+      option.dataset.tabOpened = 1 - isURLExternal(dict);
+      selectContainer.appendChild(option);
+    });
+
+    btn.addEventListener("contextmenu", (e) => {
+      e.preventDefault(); // disables default right click menu
+      selectContainer.classList.toggle("dict-select-list-hide");
+    });
+
+    btn.addEventListener("click", (e) => {
+      if (e.target === menuImgEl) return;
+      selectContainer.classList.add("dict-select-list-hide");
+    });
+
+    menuImgEl.addEventListener("click", (e) => {
+      e.stopPropagation();
+      selectContainer.classList.toggle("dict-select-list-hide");
+    });
+
+    selectButtonBox.addEventListener("mouseleave", () => {
+      selectContainer.classList.add("dict-select-list-hide");
+    });
+
+    selectContainer.addEventListener("click", (e) => {
+      const clickedOption = e.target.closest(".dict-select-option");
+      if (!clickedOption) return;
+
+      selectContainer.classList.add("dict-select-list-hide");
+
+      const optionVal = clickedOption.dataset.dictId;
+      const domain = getDictURLDomain(TERM_DICTS[optionVal]);
+      const btnLabel = domain.split("www.").splice(-1)[0];
+      const faviconURL = getFavicon(domain);
+      const faviconEl = createImg(faviconURL, "dict-btn-fav-img"); // img elements get deleted after "change" event. so we create them after each change
+      
+      btn.dataset.dictId = optionVal;
+      btn.dataset.dictExternal = clickedOption.dataset.dictExternal;
+      btn.textContent = btnLabel;
+      btn.dataset.tabOpened = clickedOption.dataset.tabOpened;
+      btn.prepend(faviconEl);
+      const menuImgEl = createImg("", "dict-btn-list-img");
+      btn.appendChild(menuImgEl);
+      
+      if (clickedOption.dataset.dictExternal == 1) {
+        loadDictPage(optionVal, "");
+
+        const arrowEl = createImg("", "dict-btn-external-img");
+        btn.appendChild(arrowEl);
+        // btn.classList.remove("dict-btn-active");
+      } else {
+        loadDictPage(optionVal, iFrame);
+        activateTab(btn, dictTabButtons);
+      }
+      // as with the icons, btn content changes so events get deleted
+      menuImgEl.addEventListener("click", (e) => {
+        e.stopPropagation();
+        selectContainer.classList.toggle("dict-select-list-hide");
+      });
+    });
+  }
+  
+  // create image button
+  const imageBtn = createTabBtn("", dictTabsContainer, -1, 0);
+  imageBtn.setAttribute("id", "dict-image-btn");
+  imageBtn.setAttribute("title", "Look up images for the term");
+  const imageFrame = createIFrame("imageframe", iFramesContainer);
+
+  dictTabButtons.set(imageBtn, imageFrame);
+  
+  // set first frame as active (for final: need to save active tab and retrieve it)
```

Comment:
Checking on the comment "need to save active tab and retrieve it" -- we don't need to save the active tab, do we?